### ISSUE TITLE
Startup fails when PERSIST_PATH directory is empty

### ISSUE THREAD

**Issue by thrau** [etspaceman/kinesis-mock#129]
Version: 0.1.2

Steps to reproduce:
* create an empty directory /tmp/kinesis-mock
* run `SHOULD_PERSIST_DATA=true PERSIST_PATH="/tmp/kinesis-mock" ./kinesis-mock-linux-amd64-static`

I would expect this to create the necessary `kinesis-data.json` in the directory, like it does when `PERSIST_PATH` is not specified. however it fails with:

```
thomas@om ~/Downloads % SHOULD_PERSIST_DATA=true PERSIST_PATH="/tmp/kinesis-mock" ./kinesis-mock-linux-amd64-static
[ioapp-compute-1] INFO  2021-06-28 20:31:26,479 k.m.KinesisMockService contextId=0c3cea9f-d83f-11eb-a703-395ba624dc2a, cacheConfig={"awsAccountId":"000000000000","awsRegion":"us-east-1","createStreamDuration":{"length":500,"unit":"MILLISECONDS"},"deleteStreamDuration":{"length":500,"unit":"MILLISECONDS"},"deregisterStreamConsumerDuration":{"length":500,"unit":"MILLISECONDS"},"initializeStreams":null,"mergeShardsDuration":{"length":500,"unit":"MILLISECONDS"},"persistConfig":{"fileName":"kinesis-data.json","interval":{"length":5,"unit":"SECONDS"},"loadIfExists":true,"path":"/tmp/kinesis-mock","shouldPersist":true},"registerStreamConsumerDuration":{"length":500,"unit":"MILLISECONDS"},"shardLimit":50,"splitShardDuration":{"length":500,"unit":"MILLISECONDS"},"startStreamEncryptionDuration":{"length":500,"unit":"MILLISECONDS"},"stopStreamEncryptionDuration":{"length":500,"unit":"MILLISECONDS"},"updateShardCountDuration":{"length":500,"unit":"MILLISECONDS"}} - Logging Cache Config 
java.io.FileNotFoundException: /tmp/kinesis-mock/kinesis-data.json (No such file or directory)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_io_FileNotFoundException_2_0002e_0003cinit_0003e_00028Ljava_lang_String_2Ljava_lang_String_2_00029V(JNIJavaCallWrappers.java:0)
	at java.io.FileInputStream.open0(FileInputStream.java)
	at java.io.FileInputStream.open(FileInputStream.java:219)
	at java.io.FileInputStream.<init>(FileInputStream.java:157)
	at com.fasterxml.jackson.core.JsonFactory.createParser(JsonFactory.java:1029)
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3076)
	at kinesis.mock.cache.Cache$.$anonfun$loadFromFile$3(Cache.scala:1247)
	at flatMap @ kinesis.mock.cache.Cache$.$anonfun$loadFromFile$2(Cache.scala:1247)
	at ifM$extension @ org.typelevel.log4cats.slf4j.internal.Slf4jLoggerInternal$.org$typelevel$log4cats$slf4j$internal$Slf4jLoggerInternal$$contextLog(Slf4jLoggerInternal.scala:55)
	at ifM$extension @ org.typelevel.log4cats.slf4j.internal.Slf4jLoggerInternal$.org$typelevel$log4cats$slf4j$internal$Slf4jLoggerInternal$$contextLog(Slf4jLoggerInternal.scala:55)
	at flatMap @ kinesis.mock.KinesisMockService$.$anonfun$run$6(KinesisMockService.scala:34)
	at ifM$extension @ org.typelevel.log4cats.slf4j.internal.Slf4jLoggerInternal$.org$typelevel$log4cats$slf4j$internal$Slf4jLoggerInternal$$contextLog(Slf4jLoggerInternal.scala:55)
	at flatMap @ kinesis.mock.KinesisMockService$.$anonfun$run$4(KinesisMockService.scala:29)
	at run @ com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at run @ com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at delay$extension @ pureconfig.module.catseffect.package$.loadF(package.scala:51)
	at subflatMap @ pureconfig.module.catseffect.package$.loadF(package.scala:52)
	at leftMap @ pureconfig.module.catseffect.package$.loadF(package.scala:53)
	at rethrowT @ pureconfig.module.catseffect.package$.loadF(package.scala:54)
	at flatMap @ kinesis.mock.KinesisMockService$.$anonfun$run$3(KinesisMockService.scala:27)
	at flatMap @ kinesis.mock.KinesisMockService$.$anonfun$run$1(KinesisMockService.scala:26)
	at run @ com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at use @ kinesis.mock.KinesisMockService$.run(KinesisMockService.scala:24)
```

this currently breaks localstack with DATA_DIR enabled.

**Comment by etspaceman** [comment]
Hey there. Unfortunately I am on vacation until next week but I can take a look when I'm back. If you want to open a PR I'd be happy to look at it.

**Comment by thrau** [comment]
No worries, enjoy your vacation. Will find a workaround in LocalStack or learn a bit of Scala tomorrow ;)

**Comment by etspaceman** [comment]
Closed w/ #133
